### DISCUSSION TITLE
可以支援其他鍵盤佈局？

### DISCUSSION THREAD

**Discussion by archerindigo** [rime/home#819] [Ideas]
現在小狼毫、ibus-rime等在ASCII模式時似乎只能支援美式鍵盤佈局，其實有沒有辦法支援其他佈局，例如US International Alt Gr？因為我有時需要混打äüöß等拉丁文字，在各個鍵盤之間切換不是很方便。

**Comment by lotem** [comment]
ibus-rime使用的佈局是在輸入法引擎組件的描述文件裏定義。
https://github.com/rime/ibus-rime/blob/d525f18b45123869af0055cc8402d459ebdbb9e3/rime.xml.in#L19
沒有研究過ibus是否支持在運行時修改鍵盤佈局。

**Reply by Bobo-alcazar** [reply]
那麼在 Windows 上呢？小狼毫可以指定關聯的 layout 嗎？

**Reply by yuxuanchen1997** [reply]
I use Colemak on windows and tried to implement this in https://github.com/rime/weasel/pull/1905

**Reply by chinanoahli** [reply]
> 那麼在 Windows 上呢？小狼毫可以指定關聯的 layout 嗎？

Windows下没有简单地修改键盘布局的办法，例如我是Dvorak用户，所有CJK输入法都默认不支持，也不能直接修改键盘布局。

在Win下实现只有三种办法：

第一是改注册表，把en-us的layout dll从kbdus.dll 改为 kbddv.dll(dvorak)对应的dll。

第二是在PE下调换两个DLL，因为只要开机，kbdus就一定会被占用。

第三是直接买一个支持dvorak布局的键盘。

要么就是修改软件源码。

其中简体中文对应的注册表路径是：`HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\Keyboard Layouts\00000804`把`Layout File`的值从`KBDUS.DLL`改为你需要的键盘布局dll，然后重启即可

其中 00000804 为语言代码，如果是非中国大陆简体中文的系统，需要到微软文档查询对应语言的代码。

> [!WARNING] 
>
> 不管是改注册表还是暴力地调换两个DLL的名称，都需要在执行前先移除全部用户账户的密码。
> 
> 修改之后，会导致用户账户密码输入识别错乱。
>
> 修改后需要重启才能生效。

键盘布局对应的DLL可以在微软文档里面查询，也可以在`system32`里面找到任何`kb*.dll`并查询其属性。

<img width="976" height="505" alt="截圖 2026-08-15 晚上10 21 08" src="https://github.com/user-attachments/assets/b53c5963-ff22-4c2a-92b9-d0fd5f71210d" />
